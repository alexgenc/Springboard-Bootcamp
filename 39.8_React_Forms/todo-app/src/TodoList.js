import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import { v4 as uuidv4 } from 'uuid';

const TodoList = () => {

  const [todos, setTodos] = useState([]);

  const addTodo = (todo) => {
    let newTodo = {...todo, id: uuidv4()}
    setTodos( (todos) => [...todos, newTodo]);
  }

  const removeTodo = (id) => {
    setTodos( (todos) => todos.filter(todo => todo.id !== id ));
  };

  const allTodos = todos.map(todo => (
    <Todo 
      key = {todo.id}
      id = {todo.id}
      task = {todo.task}
      handleRemove = {removeTodo}
    />
  ));

  return (
    <div>
      <div>
        <NewTodoForm addTodo={addTodo} />
      </div>  
      <div>
        {allTodos}
      </div>  
    </div>
  )
}

export default TodoList;