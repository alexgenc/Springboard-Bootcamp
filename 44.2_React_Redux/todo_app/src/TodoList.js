import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

const TodoList = () => {
  
  const dispatch = useDispatch();
  const todos = useSelector(store => store.todos);
  
  const handleCreate = (task) => {
    dispatch({
      type: 'ADD_TODO',
      task
    });
  }

  const handleDelete = (id) => {
    dispatch({
      type: 'REMOVE_TODO',
      id
    });
  }

  const allTodos = todos.map(todo => (
    <Todo 
      deleteTodo={handleDelete}
      id={todo.id}
      key={todo.id}
      task={todo.task}
    />
  ));

  return (
    <div>
      <hr />
      <NewTodoForm createTodo={handleCreate} />
      <hr />
      <ul>{allTodos}</ul>
    </div>
  )

  
}

export default TodoList;