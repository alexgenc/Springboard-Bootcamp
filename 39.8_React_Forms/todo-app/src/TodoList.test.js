import React from "react";
import { render } from "@testing-library/react";
import TodoList from "./TodoList";

// Smoke Test
it("renders without crashing", () => {
  render(<TodoList />);
});

// Snapshot Test
it("matches snapshot", () => {
  const {asFragment} = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});