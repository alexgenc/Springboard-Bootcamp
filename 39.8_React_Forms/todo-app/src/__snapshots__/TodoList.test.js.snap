// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <h2>
          Add a Todo
        </h2>
        <form>
          <div>
            <label
              for="task"
            >
              Task:
            </label>
            <input
              id="task"
              name="task"
              value=""
            />
          </div>
          <button>
            Add Todo
          </button>
        </form>
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
